tee_supplicant/src/tee_supplicant.c

```diff
@@ -52,6 +72,100 @@ static int receive_request(const struct device *dev, struct tee_supp_msg *ts_req
 	return rc;
 }
 
+static struct suppl_shm *find_shm(uint64_t id)
+{
+	struct suppl_shm *curr = NULL;
+	uint64_t mem_id;
+
+	if (id) {
+		mem_id = (uint64_t)((struct tee_shm *)id)->addr;
+
+		tee_supp_mutex_lock(&shm_mutex);
+
+		curr = shm_head;
+		while (curr && curr->id != mem_id) {
+			curr = curr->next;
+		}
+
+		tee_supp_mutex_unlock(&shm_mutex);
+	}
+
+	return curr;
+}
+
+static struct suppl_shm *pop_shm(uint64_t id)
```

Comment:
"pop" implies stack-like operation, which is not the case. How about "forget_shm"?

Reply:
renamed